vertx-junit5/src/main/java/io/vertx/junit5/CountingCheckpoint.java

```diff
@@ -47,11 +48,22 @@ private CountingCheckpoint(Consumer<Checkpoint> satisfactionTrigger, Consumer<Th
     if (requiredNumberOfPasses <= 0) {
       throw new IllegalArgumentException("A checkpoint needs at least 1 pass");
     }
+    this.creationCallSite = findCallSite();
     this.satisfactionTrigger = satisfactionTrigger;
     this.overuseTrigger = overuseTrigger;
     this.requiredNumberOfPasses = requiredNumberOfPasses;
   }
 
+  private StackTraceElement findCallSite() {
+    StackTraceElement[] stackTrace = new Exception().getStackTrace();
```

Comment:
Why this statement more than https://github.com/vert-x3/vertx-junit5/blob/master/vertx-junit5-web-client/src/main/java/io/vertx/junit5/web/TestRequest.java#L289 `Thread.currentThread().getStackTrace()`? Any difference?

Reply:
The goal here is to just pin-point the flag creation line in the code. The full stacktrace is not so much relevant since there is no execution context of interest, you really just want to identify which checkpoints have been left unsatisfied.